octree/include/pcl/octree/impl/octree_iterator.hpp

```diff
@@ -281,6 +281,56 @@ namespace pcl
 
       return (*this);
     }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeBreadthFirstAtDepthIterator<OctreeT>::OctreeBreadthFirstAtDepthIterator (unsigned int depth_arg) :
+        OctreeBreadthFirstIterator<OctreeT> (depth_arg), depth_(depth_arg)
+    {
+      OctreeBreadthFirstIterator<OctreeT>::reset ();
```

Comment:
`OctreeBreadthFirstAtDepthIterator<OctreeT>::reset()` already calls `OctreeBreadthFirstIterator<OctreeT>::reset ()` so this is useless.